### PR TITLE
Add YaoiScan

### PR DESCRIPTION
**Pull request by FunnyTiming**
Checklist:

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [ ] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/fr/yaoiscan/src/YaoiScan.kt [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+class YaoiScan : MangaReader() {
+
+    override val name = "YaoiScan"
+
+    override val lang = "fr"
+
+    override val baseUrl = "https://yaoiscan.fr/"
+
+    override val client = network.cloudflareClient.newBuilder()
+        // .rateLimit(2)
```
Comment (FunnyTiming) [inline src/fr/yaoiscan/src/YaoiScan.kt]: I have some questions about this. Is it a Cloudflare client by default? (And so do I need to remove the override?)

Comment (bapeey) [reply src/fr/yaoiscan/src/YaoiScan.kt]: [Multisrc](<https://github.com/keiyoushi/extensions-source/blob/f15e50aa0a69bce289fd7d8f01b8620315870e7f/lib-multisrc/mangareader/src/eu/kanade/tachiyomi/multisrc/mangareader/MangaReader.kt#L25>) is already using `cloudflareClient` by default, so you should use `super.client` if you want to override it (`rateLimit` is comented?)

**Comment by AwkwardPeak7** [comment]
The site seems to be using `MangaThemesia` multisrc

**Comment by FunnyTiming** [comment]
> The site seems to be using `MangaThemesia` multisrc

Noooo, really ? It was soo long, but there is nothing clear to say "Hey, I'm this theme / CMS"... :skull:

**Comment by vetleledaal** [comment]
> there is nothing clear to say "Hey, I'm this theme / CMS"... 💀

You kinda have to poke around, se what query keys are used, how the 404 page looks like, and such. For MangaThemesia it's especially annoying since people keep renaming the theme internally -- so you can't blindly trust the name in WordPress.

I have been using this, where anything >50% is likely to be MangaThemesia:
```js
($$("#main-menu, .searchx, .postbody, .blox.kln, #content, header, .logos, span.hdl, .shme").length*100/9).toFixed(0)+"%"
```

The other common one you will run into is Madara.

**Comment by FunnyTiming** [comment]
Sorry for the commit history... Committer was not myself so...

File: src/fr/yaoiscan/build.gradle [hunk 2]
```diff
@@ -0,0 +1,10 @@
+ext {
+    extName = 'YaoiScan'
+    extClass = '.YaoiScan'
+    themePkg = 'mangathemesia'
+    baseUrl = 'https://yaoiscan.fr/'
```
Comment (vetleledaal) [inline src/fr/yaoiscan/build.gradle]: Remove trailing slash
```suggestion
    baseUrl = 'https://yaoiscan.fr'
```

**Review by vetleledaal — APPROVED** [review]

**Pull request merged**